db/compaction/compaction_picker_level.cc

```diff
@@ -273,6 +276,13 @@ void LevelCompactionBuilder::SetupInitialFiles() {
     compaction_reason_ = CompactionReason::kForcedBlobGC;
     return;
   }
+
+  // Files pointed by compact cursor in kRoundRobin should respect boosted ttl
```

Comment:
I'm not sure about that, that RoundRobin (RR) should respect boosted ttl. I think we may need more data to support that. (I think ttl_boost is for the level compaction to prioritize old SST to lower level, which RR already naturally has.)
I think we should improve the TTL compaction for RR, which is picks the old files from the last-1 level. For RR LSM tree time, in general it should have
1. in each level, the timestamp of each SST is sorted, the first file after the compaction cursor is the oldest (the one before the cursor is the newest);
2. the timestamp for each level is also sorted and have no time range overlap, for example, L2 oldest file (after cursor) > L3 newest file (> means newer).
Then TTL compaction should just pick the files from the "tail" of the LSM tree (the cursor of the last - 1 level) until all expired files are picked. (so SST timestamp wise, it's more like a FIFO compaction **in general**, it's not strictly true I think)

Then the idea of ttl boost might hurt the WA instead of improving it, because the extra high level compaction triggered by ttl_booster will only make more files are eligible for future TTL compaction. for example, in the following diagram, `T` is time, the larger it is, the younger the file is. If `T20` is boosted and compacted to L3, more old files are generated:

![image](https://user-images.githubusercontent.com/33968/181421150-6bc81b36-1520-40c6-94d0-1d7ababe91f9.png)

Compaction output files are likely be all old timestamp:
![image](https://user-images.githubusercontent.com/33968/181422401-436a5fa1-d182-47f8-b21d-b53ce4782c7c.png)

I think we might want to do some tests to check that.

Reply:
Thanks for your detailed analysis. You're right. It is still unclear if this can give us benefit. I'm running light benchmark to check it.